Applications/DataHolderLib/BoundaryCondition.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *
+ */
+
+#pragma once
+#include "Applications/DataHolderLib/FemCondition.h"
+
+namespace DataHolderLib
+{
+class BoundaryCondition : public DataHolderLib::FemCondition
+{
+public:
+    enum ConditionType
```

Comment:
I'm not sure, if you rely on implicit conversion of the enum to integer types, but if not, please make the enum an `enum class`. If you rely on the implicit conversions, consider to add explicit type conversions.